src/core/operations_problem.jl

```diff
@@ -305,8 +307,17 @@ function construct_device!(
     if haskey(op_problem.template.devices, name)
         throw(IS.ConflictingInputsError("Device with model name $(name) already exists in the Opertaion Model"))
     end
-    devices_ref = get_devices_ref(op_problem)
-    devices_ref[name] = device_model
+    if device_model.device_type <: PSY.Branch
```

Comment:
we should split these into function and call them `set_model(::PSY.Type)`

Reply:
I don't follow this suggestion. I could see a function with seveeral methods like: `get_component_ref(<:PSY.Branch)` that returns the field of the template so that the addition is made in the right place.

Reply:
So, just to be clear... you are suggesting something like:
```julia
function set_model(op_problem::OperationsProblem, name::Symbol, device_type::D, device_model::DeviceModel) where D <: PSY.Branch
    branches_ref = get_branches_ref(op_problem)
    branches_ref[name] = device_model
end

function set_model(op_problem::OperationsProblem, name::Symbol, device_type::D, device_model::DeviceModel) where D <: PSY.StaticInjection
    devices_ref = get_devices_ref(op_problem)
    devices_ref[name] = device_model
end

function set_model(op_problem::OperationsProblem, name::Symbol, device_type::D, device_model::DeviceModel) where D <: PSY.Service
    services_ref = get_services_ref(op_problem)
    services_ref[name] = device_model
end
```
@jd-lara is that correct?
I don't know if this is an improvement. Especially since this won't have a performance impact.

Reply:
This will allow us to keep flexible the internal implementation of the template.

Reply:
nit:  Julia convention is to prioritize function parameters that are types.
```
function set_model(::PSY.StaticInjection, op_problem, name, device_model)
```

Also, since you are in operations_problem.jl I wouldn't hesitate to use 

```
op_problem.template.devices[name] = device_model
```